python/ql/lib/semmle/python/dataflow/new/internal/DataFlowDispatch.qll

```diff
@@ -1737,7 +1811,10 @@ abstract class ReturnNode extends Node {
 /** A data flow node that represents a value returned by a callable. */
 class ExtractedReturnNode extends ReturnNode, CfgNode {
   // See `TaintTrackingImplementation::returnFlowStep`
-  ExtractedReturnNode() { node = any(Return ret).getValue().getAFlowNode() }
+  ExtractedReturnNode() {
+    node = any(Return ret).getValue().getAFlowNode() or
+    node = any(Yield yield).getAFlowNode()
```

Comment:
With the new yield store step, where the yielded element is stored to the `yield` expression, it seems fine to simply use the yield as a normal return. We are returning a new thing and we can control its content. I have seen C# use a different return kind for variables that already exist but are written to.

Reply:
I understood that as: since I got `yield` working without messing with the ReturnKinds, I didn't really investigate that path much.

I guess that's fine :+1: I'm still a little curious how the approach would have worked out :thinking: (but it's not super important so :shrug:)